rust/chains/hyperlane-cardano/src/bin/cardano_rpc_main.rs

```diff
@@ -0,0 +1,59 @@
+use hyperlane_cardano::rpc::CardanoRpc;
+use hyperlane_cardano::{
+    CardanoMailbox, CardanoMailboxIndexer, CardanoValidatorAnnounce, ConnectionConf,
+};
+use hyperlane_core::{
+    ChainResult, ContractLocator, HyperlaneDomain, HyperlaneMessage, IndexRange, Indexer,
+    KnownHyperlaneDomain, Mailbox, ValidatorAnnounce, H256,
+};
+use std::str::FromStr;
+
+#[tokio::main(flavor = "current_thread")]
+async fn main() -> ChainResult<()> {
+    let cardano_rpc = CardanoRpc::new(&"http://localhost:3000".parse().unwrap());
+    let finalized_block_number = cardano_rpc.get_finalized_block_number().await.unwrap();
```

Comment:
Can we minimise the use of `unwraps./ panics` and handle errors gracefully?

Reply:
While I agree, I think we can do this after the core functionalities are done, then do tight error handling in a unified PR. Doing it now adds a lot of friction to development, and risks a redo anyway as we decide on an error handling "standard". `unwrap`s are overused in the original Hyperlane codebase to start with...

cc @serejke.

Reply:
Yes, error handling will be improved later: https://github.com/tvl-labs/hyperlane-cardano/issues/59

Reply:
Thanks, although , that would only cover the rpc , and we might want to evaluate better error handling through out the agent.